scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/ExecuterInformationData.java

```diff
@@ -98,17 +105,39 @@ public ExecuterInformation toExecuterInformation(boolean loadFullState) {
     }
 
     private TaskLauncher getReboundTaskLauncherIfStillExist() {
-        try {
-            logger.debug("List AOs on " + taskLauncherNodeUrl + " (expect only one): " +
-                         Arrays.toString(NodeFactory.getNode(taskLauncherNodeUrl).getActiveObjects()));
-            Object[] aos = NodeFactory.getNode(taskLauncherNodeUrl).getActiveObjects();
-            return (TaskLauncher) aos[0];
-        } catch (Throwable t) {
-            logger.warn("Failed to rebind TaskLauncher of task " + taskId + ". TaskLauncher with node URL: " +
-                        taskLauncherNodeUrl +
+        TaskLauncher answer = null;
+        boolean nodeConnected = false;
+        boolean isPAMR = taskLauncherNodeUrl.startsWith("pamr://");
+        // When using the pamr protocol, nodes (pamr agents) will reconnect by re-registering with the router using the same agent id.
+        // Each reconnection attempt is using a maximum interval of 64 seconds, the router will not know of these agents until the reconnection occurs.
+        long initialTime = System.currentTimeMillis();
+        String nodeUrl = nodes.get(0).getNodeInformation().getURL();
+        do {
+            try {
+                logger.debug("List AOs on " + nodeUrl + " (expect only one): ");
+                Object[] aos = NodeFactory.getNode(nodeUrl).getActiveObjects();
+                nodeConnected = true;
+                logger.debug(Arrays.toString(aos));
+                answer = (TaskLauncher) aos[0];
+            } catch (NodeException t) {
+                if (isPAMR) {
+                    logger.debug("Failed to access node " + nodeUrl);
+                    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+                }
+            } catch (Throwable t) {
+                // task launcher does not exist (any more)
+            }
+            // Once the maximum pamr reconnection interval has been reached, no agent id re-attribution should occur,
+            // which means the nodes are definitely not reachable
+            pamrReconnectionTimeoutSpent = pamrReconnectionTimeoutSpent ||
+                                           (System.currentTimeMillis() - initialTime > 64000);
```

Comment:
Could you use extract constant here? why it is 64 seconds?

Reply:
This is hardcoded inside PAMR protocol:
https://github.com/ow2-proactive/programming/blob/proactive-light/programming-extensions/programming-extension-pamr/src/main/java/org/objectweb/proactive/extensions/pamr/client/AgentImpl.java
Method geTunnelOrReconnect:
if (delay < 1000 * 60) {
         delay *= 2;
 }

Reply:
I created a PR in programming to define a constant
https://github.com/ow2-proactive/programming/pull/1367

Reduced the maximum delay a bit